dashboard/src/hooks/use-hash-key.ts

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+
+export function useHashKey(key: string): [boolean, (state: boolean) => void] {
+  const router = useRouter();
+  const [isKeySet, setIsKeySet] = useState(false);
+
+  useEffect(() => {
+    const checkHash = () => {
+      if (typeof window !== 'undefined') {
+        setIsKeySet(window.location.hash === `#${key}`);
+      }
+    };
+    checkHash();
+    window.addEventListener('hashchange', checkHash);
+    return () => {
+      window.removeEventListener('hashchange', checkHash);
+    };
+  }, [key]);
+
+  const toggleKey = (state: boolean) => {
+    const hash = state ? `#${key}` : '';
+    router.push(`${window.location.search}${hash}`);
+    setIsKeySet(state);
+  };
```

Comment:
I've just tested it again after seeing this.
It looks like this causes it to a perform a full page reload/refresh, every time you add / remove the hash key.
I think that by pushing a URL to the router like this, it handles it as a redirect, causing the entire page to have to be fetched and loaded.